### PR TITLE
chore(oas): replace response with $ref class JSDoc (Store PRO-V)

### PR DESCRIPTION
**Pull request by patrick-medusajs**
### Scope

Store routes directories PRO to V.

### What

Move inline OAS response schema declaration under their respective class declarations in order to expose them through  `#/components/schemas`. Replace inline OAS response schema with a `$ref` reference pointing to the newly declared schema.

### Why

Having response declared as its own "named" schema will allow OAS code generators to output typed entities/DTO that can be consumed without having to reference the route/operation.

### How

Declare a new @schema JSDoc for each "Res" class used to parse and validate request body. Move the current inline requestBody to the new @schema.

### Test

- Ran OAS validator.
- Ran docs build script.

Expect no visible changes to the documentation.

### CODE REVIEW

File: packages/medusa/src/api/routes/store/product-categories/index.ts [hunk 1]
```diff
@@ -49,4 +50,15 @@ export const allowedStoreProductCategoryFields = [
...
+/**
+ * @schema StoreGetProductCategoryRes
+ * type: object
+ * properties:
+ *   product_category:
+ *     $ref: "#/components/schemas/ProductCategory"
+ */
+export type StoreGetProductCategoryRes = {
+  product_category: ProductCategory
+}
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/product-categories/index.ts:62]: Added missing declaration.

File: packages/medusa/src/api/routes/admin/products/index.ts [hunk 2]
```diff
@@ -211,7 +211,9 @@ export type AdminProductsDeleteRes = {
  *   products:
  *     type: array
  *     items:
- *       $ref: "#/components/schemas/Product"
+ *       oneOf:
+ *         - $ref: "#/components/schemas/Product"
+ *         - $ref: "#/components/schemas/PricedProduct"
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/products/index.ts:216]: Now matches the typed response.
https://github.com/medusajs/medusa/blob/4424301ae27ea0dbcd97bb284462e56ce2a52698/packages/medusa/src/api/routes/admin/products/index.ts#L227-L229

File: packages/medusa/src/api/routes/store/products/list-products.ts [hunk 3]
```diff
@@ -151,30 +151,7 @@ import PublishableAPIKeysFeatureFlag from "../../../../loaders/feature-flags/pub
...
- *                 allOf:
- *                   - $ref: "#/components/schemas/Product"
- *                   - type: object
- *                     properties:
- *                       variants:
- *                         type: array
- *                         items:
- *                           allOf:
- *                             - $ref: "#/components/schemas/ProductVariant"
- *                             - $ref: "#/components/schemas/ProductVariantPricesFields"
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/products/list-products.ts:168]: Replace with PricedProduct

File: packages/medusa/src/api/routes/store/variants/get-variant.ts [hunk 4]
```diff
@@ -41,12 +41,7 @@ import { validator } from "../../../../utils/validator"
  *     content:
  *       application/json:
  *         schema:
- *           type: object
- *           properties:
- *             variant:
- *               allOf:
- *                 - $ref: "#/components/schemas/ProductVariant"
- *                 - $ref: "#/components/schemas/ProductVariantPricesFields"
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/variants/get-variant.ts:49]: Replace with PricedVariant.
https://github.com/medusajs/medusa/blob/4424301ae27ea0dbcd97bb284462e56ce2a52698/packages/medusa/src/types/pricing.ts#L37-L79

File: packages/medusa/src/api/routes/store/variants/index.ts [hunk 5]
```diff
@@ -16,11 +15,27 @@ export default (app) => {
...
+/**
+ * @schema StoreVariantsListRes
+ * type: object
+ * properties:
+ *   variants:
+ *     type: array
+ *     items:
+ *       $ref: "#/components/schemas/PricedVariant"
+ */
+export type StoreVariantsListRes = {
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/variants/index.ts:38]: Endpoint does not return PaginatedResponse fields. Removed from typing.

File: packages/medusa/src/api/routes/store/variants/list-variants.ts [hunk 6]
```diff
@@ -73,14 +73,7 @@ import { IsType } from "../../../../utils/validators/is-type"
...
  *       application/json:
  *         schema:
- *           type: object
- *           properties:
- *             variants:
- *               type: array
- *               items:
- *                 allOf:
- *                   - $ref: "#/components/schemas/ProductVariant"
- *                   - $ref: "#/components/schemas/ProductVariantPricesFields"
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/variants/list-variants.ts:83]: Replace with PricedVariant

File: packages/medusa/src/models/product-variant.ts [hunk 7]
```diff
@@ -216,44 +216,3 @@ export class ProductVariant extends SoftDeletableEntity {
  *     description: An optional key-value map with additional details
  *     example: {car: "white"}
  */
-
-/**
- * @schema ProductVariantPricesFields
```
Comment (patrick-medusajs) [inline packages/medusa/src/models/product-variant.ts:221]: OAS was folded under PricedVariant

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting noteworthy code changes.

Should also resolve DOCS-217

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Review by fPolic — APPROVED** [review]
LGTM!

File: packages/medusa/src/api/routes/store/product-types/index.ts [hunk 8]
```diff
@@ -39,12 +39,24 @@ export const defaultStoreProductTypeFields = [
 ]
 export const defaultStoreProductTypeRelations = []
 
+/**
+ * @schema StoreProductTypesListRes
+ * type: object
+ * properties:
+ *   product_types:
+ *     $ref: "#/components/schemas/ProductType"
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/product-types/index.ts:47]: Should be an array

**Review by shahednasser — CHANGES_REQUESTED** [review]
Just a small comment 🙂

**Review by pKorsholm — APPROVED** [review]
LGTM

**Review by shahednasser — APPROVED** [review]

**Pull request merged**